nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/test/java/org/apache/nifi/processors/salesforce/PutSalesforceRecordIT.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.nifi.util.MockComponentLog;
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class PutSalesforceRecordIT implements SalesforceConfigAware {
+
+    private TestRunner runner;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        int maxRecordCount = 2;
+        Processor putSalesforceRecord = new CustomPutSalesforceRecord(maxRecordCount);
+
+        runner = TestRunners.newTestRunner(putSalesforceRecord);
+
+        StandardOauth2AccessTokenProvider oauth2AccessTokenProvider = initOAuth2AccessTokenProvider(runner);
+        runner.setProperty(QuerySalesforceObject.TOKEN_PROVIDER, oauth2AccessTokenProvider.getIdentifier());
+    }
+
+    @Test
+    void testPutSalesforceRecord() throws Exception {
+        MockComponentLog mockComponentLog = new MockComponentLog("id1", "testPutSalesforceRecord");
+        InputStream in = readFile("src/test/resources/json/put_records.json");
+
+        MockFlowFile flowFile = new MockFlowFile(1L);
+        byte[] fileContent = Files.readAllBytes(Paths.get("src/test/resources/json/put_records.json"));
+        flowFile.setData(fileContent);
+        flowFile.putAttributes(Collections.singletonMap("objectType", "Account"));
+
+        MockRecordParser reader = new MockRecordParser();
+        reader.addSchemaField("name", RecordFieldType.STRING);
+        reader.addSchemaField("phone", RecordFieldType.STRING);
+        reader.addSchemaField("website", RecordFieldType.STRING);
+        reader.addSchemaField("numberOfEmployees", RecordFieldType.STRING);
+        reader.addSchemaField("industry", RecordFieldType.STRING);
+
+        reader.addRecord("SampleAccount1", "111111", "www.salesforce1.com", "100", "Banking");
+        reader.addRecord("SampleAccount2", "222222", "www.salesforce2.com", "200", "Banking");
+        reader.addRecord("SampleAccount3", "333333", "www.salesforce3.com", "300", "Banking");
+        reader.addRecord("SampleAccount4", "444444", "www.salesforce4.com", "400", "Banking");
+        reader.addRecord("SampleAccount5", "555555", "www.salesforce5.com", "500", "Banking");
+
+        reader.createRecordReader(flowFile, in, mockComponentLog);
+
+        runner.addControllerService("reader", reader);
+        runner.enableControllerService(reader);
+
+        runner.setProperty(PutSalesforceRecord.API_VERSION, VERSION);
+        runner.setProperty(PutSalesforceRecord.API_URL, BASE_URL);
+        runner.setProperty(PutSalesforceRecord.RECORD_READER_FACTORY, reader.getIdentifier());
+
+
+        runner.enqueue(flowFile);

```

Comment:
With my previous suggestion.
```suggestion
        runner.enqueue("", Collections.singletonMap("objectType", "Account"));

```